packages/scoutgame-ui/src/providers/PurchaseProvider.tsx

```diff
@@ -66,11 +65,11 @@ export function PurchaseProvider({ children }: { children: ReactNode }) {
     executeAsync: checkDecentTransaction
   } = useAction(checkDecentTransactionAction, {
     onSuccess({ input }) {
-      showMessage(`Transaction ${input.txHash || ''} was successful`, 'success');
+      toast.success(`Transaction ${input.txHash || ''} was successful`);
```

Comment:
ideally, I thought we'd use sonner's loading toast to show the state here: https://sonner.emilkowal.ski/toast#promise. I think to do it properly, you would capture the promise returned from checkDecentTransaction. You could wrap the method inside of here so the components don't have to do it:
```
const checkDecentTransactionWrapped = (args: any) => {
  const decentTxPromise =  checkDecentTransaction(args);
  toast.promise(decentTxPromise, {
    loading: 'Loading...',
    success: (data) => {
      return `${data.name} toast has been added`;
    },
    error: 'Error',
  });
});
``

Reply:
Good catch. I didn't know we have that capability.

I thought of a simple implementation which I pushed:

```
const checkResultPromise = checkDecentTransaction({
    pendingTransactionId: res.data.id,
    txHash: res.data.txHash
});

toast.promise(checkResultPromise, {
  loading: 'Transaction is being settled...',
  success: () => `Transaction ${res?.data?.txHash || ''} was successful`,
  error: (data) => `Transaction failed: ${data?.serverError?.message || 'Something went wrong'}`
});

const checkResult = await checkResultPromise;

.... Rest is the same
```

Reply:
lgtm